src/components/Card.js

```diff
@@ -18,12 +18,13 @@ export const Card = () => {
     const needsEscape = ['"', ';', ',', ':', '\\'];
 
     let escaped = '';
-    for (let i = 0; i < v.length; i++) {
-      let c = v[i];
+
+    for (let c of v) {
+      let tmp = c;
```

Comment:
addidtional variable _tmp_, because `loop counters should not be modified in the body of the loop. However other loop control variables representing logical values may be modified in the loop, for example a flag to indicate that something has been completed, which is then tested in the for statement.`

Reply:
To be honest, I'm not super concerned with what SonarLint thinks. I just use Prettier so the formatting is consistent across contributes and leave the code quality/implementation to experience and opinion.

In this specific case, `c` isn't a _counter_ or index, it's the value at the index. `c` is short of `char` but that's a reserved name. Either way, modifying it is at best not ideal and probably unexpected. But I'm also not a big fan of the ambiguous `tmp`.

I think the following would be a better implementation:
- Uses `const` instead of `let` for `c` so it's immutable
- Uses string templating instead of concatenation

```js
for (const c of v) {
  if (needsEscape.includes(c) {
    escaped += `\\${c}`
  } else {
    escaped += c
  }
}
```

The original implementation purposely only modified `escaped` in one place, but adding the else block keeps it all contained and seems like a fair trade off. What do you think?

Reply:
It looks cool now. There is less code, fewer variables, and decoding is easier to understand what's going on. 
Your view of the whole thing is interesting.  👍 👍 
PR updated.